meteor/lib/rundown/rundownTiming.ts

```diff
@@ -436,9 +436,15 @@ export class RundownTimingCalculator {
 
 	private getRundownsBeforeNextBreak(
 		orderedRundowns: Rundown[],
-		currentRundown: Rundown | undefined
+		currentRundown: Rundown | undefined,
+		playlist: RundownPlaylist
 	): BreakProps | undefined {
-		const currentState = orderedRundowns.map((r) => r.endOfRundownIsShowBreak ?? '_').join('')
+		const currentState = JSON.stringify(orderedRundowns.map((r) => ({
```

Comment:
Stringifying objects is very slow, it might be better to store an array of these objects and use something like `_.isEqual` to check for equivalence. The original implementation used a string because there's not much data, this feels like something that deserves it's own type definition so we can verify which data is being used to check the cache validity.

Reply:
I have removed `JSON.stringify`, such that the currentState is a list of the objects. The type has been updated in its own interface and `_.isEqual` is being used for the deep comparison to trigger the recalculation.